consumer/consumer.go

```diff
@@ -27,16 +27,17 @@ const (
 	// commitModulo is used to define the batch size of offsets based on the current offset being processed
 	commitModulo = 10
 
-	OperationTypeCreated   = "created"
-	OperationTypeUpdated   = "updated"
-	OperationTypeDeleted   = "deleted"
-	OperationTypeMigration = "migration"
+	OperationTypeReportResource = "ReportResource"
+	OperationTypeDeleteResource = "DeleteResource"
+	OperationTypeMigration      = "migration"
 )
 
 var (
-	requiredHeaders = []string{"operation"}
-	ErrClosed       = errors.New("consumer closed")
-	ErrMaxRetries   = errors.New("max retries reached")
+	requiredHeaders  = []string{"operation", "version"}
+	validOperations  = []string{OperationTypeReportResource, OperationTypeDeleteResource, OperationTypeMigration}
+	validApiVersions = []string{"v1beta2"}
```

Comment:
We could consider using a map instead of a slice for these. It would allow an O(1) performance on lookups instead of using `slice.contains` at the cost of some memory.

This might fall under future enhancement though, because a map would also be better for coupling API versions to specific methods.

Reply:
+1 to maps, much cleaner code and logic, and avoids pulling in strings pkg for just a few validation checks! Updated to use maps instead